django/template/base.py

```diff
@@ -954,14 +954,10 @@ class NodeList(list):
     contains_nontext = False
 
     def render(self, context):
-        bits = []
-        for node in self:
-            if isinstance(node, Node):
-                bit = node.render_annotated(context)
-            else:
-                bit = node
```

Comment:
The one place this branch might happen is if someone has used `register.tag` to register a tag that doesn't return a `Node` instance. It makes pretty clear in the docs that you should return a `Node`, but it feels like that someone could very easily fall into the trap of not doing so. Since (up until now) such an error would still work (I think) the error would go undetected.

As such, maybe a release note could be warranted? It would mean that if some tags suddenly break upon updating Django, then the release notes would provide an avenue for investigation. On the other hand, perhaps I am overestimating how likely this is. 🤷

Reply:
> As such, maybe a release note could be warranted?

IMO release notes are not necessary. I don't think that registering a tag that doesn't return a `Node` instance works now, it should crash on nodes compilation 
https://github.com/django/django/blob/de4f6201835043ba664e8dcbdceffc4b0955ce29/django/template/base.py#L158
e.g. with an attribute error like `...has no attribute 'must_be_first'`.